dirty_cat/datasets/fetching.py

```diff
@@ -1,493 +1,380 @@
-
-    # chunk writing is not implemented, see if necessary
-    if _check_if_exists(temp_full_name, remove=False):
-        if md5sum is not None:
-            if _md5_sum_file(temp_full_name) != md5sum:
-                raise FileChangedError(
-                    "File %s checksum verification has failed."
-                    "Dataset fetching aborted." % temp_full_name)
-
-        shutil.move(temp_full_name, full_name)
-    if _check_if_exists(full_name, remove=False) and uncompress:
-        _uncompress_file(full_name, delete_archive=True)
-
-    if download and (initial_encoding != 'utf-8'):
-        for file in os.listdir(data_dir):
-            _change_file_encoding(
-                os.path.join(data_dir, file), initial_encoding, 'utf-8')
-    return full_name
-
-
-def fetch_employee_salaries():
-    """fetches the employee_salaries dataset
-
-    The employee_salaries dataset contains information about annual salaries
-    (year 2016) for more than 9,000 employees of the Montgomery County
-    (Maryland, US).
+from dirty_cat.datasets.utils import get_data_dir
 
 
-    Returns
-    -------
-    dict
-        a dictionary containing:
+Details = namedtuple("Details", ["name", "file_id", "description"])
+Features = namedtuple("Features", ["names"])
 
-            - a short description of the dataset (under the ``DESCR``
-              key)
-            - the tabular data (under the ``data`` key)
-            - the target (under the ``target`` key)
+# Directory where the ``.gz`` files containing the
+# details on downloaded datasets are stored.
+# Note: the tree structure is created by ``fetch_openml()``.
+# As of october 2020, this function is annotated as
+# ``Experimental`` so the structure might change in future releases.
+# This path will be concatenated to the dirty_cat data directory,
+# available via the function ``get_data_dir()``.
+DETAILS_DIRECTORY = "openml/openml.org/api/v1/json/data/"
 
-    References
-    ----------
-    https://catalog.data.gov/dataset/employee-salaries-2016
+# Same as above ; for the datasets features location.
+FEATURES_DIRECTORY = "openml/openml.org/api/v1/json/data/features/"
 
-    """
+# Same as above ; for the datasets data location.
+DATA_DIRECTORY = "openml/openml.org/data/v1/download/"
 
-    data = fetch_openml(data_id=42125, as_frame=True)
-    data.data['Current Annual Salary'] = data['target']
-    return data
+# The IDs of the datasets, from OpenML.
+# For each dataset, its URL is constructed as follows:
+openml_url = "https://www.openml.org/d/{ID}"
+ROAD_SAFETY_ID = 42803
+OPEN_PAYMENTS_ID = 42738
+MIDWEST_SURVEY_ID = 42805
+MEDICAL_CHARGE_ID = 42720
+EMPLOYEE_SALARIES_ID = 42125
+TRAFFIC_VIOLATIONS_ID = 42132
+DRUG_DIRECTORY_ID = 43044
 
-    # link dead.
-    # return fetch_dataset(EMPLOYEE_SALARIES_CONFIG, show_progress=False)
 
+def fetch_openml_dataset(dataset_id: int, data_directory: Path = get_data_dir()) -> dict:
```

Comment:
I think that this line has a couple of PEP8 violations:
- too long
- spaces in front and after the equal sign

You should consider using a pep8 violation highlighter such as flake8